src/common/src/buffer/bitmap.rs

```diff
@@ -631,18 +637,31 @@ unsafe impl TrustedLen for BitmapIter<'_> {}
 
 pub struct BitmapOnesIter<'a> {
     bitmap: &'a Bitmap,
-    idx: usize,
+    cur_idx: usize,
+    cur_bits: Option<usize>,
 }
 
 impl<'a> iter::Iterator for BitmapOnesIter<'a> {
     type Item = usize;
 
     fn next(&mut self) -> Option<Self::Item> {
-        let next_idx = self.bitmap.next_set_bit(self.idx);
-        if let Some(idx) = next_idx {
-            self.idx = idx + 1;
+        while self.cur_bits == Some(0) {
+            self.cur_idx += 1;
+            self.cur_bits = if self.cur_idx == (self.bitmap.len() + BITS - 1) / BITS {
```

Comment:
A little tricky but I think it is ok
```suggestion
            self.cur_bits = if self.cur_idx >= self.bits.len() {
```